client/src/components/ui/QRScanner.svelte

```diff
@@ -0,0 +1,108 @@
+<script lang="ts">
+    import QrScanner from 'qr-scanner';
+    import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+    import { Events, ButtonType } from '../types';
+    import { assert } from '../../assert';
+    import { Document, DocumentSchema } from '../../../../model/src/document.ts';
+
+    import { topToastMessage } from '../../pages/dashboard/stores/ToastStore';
+    import Button from './Button.svelte';
+
+    export let maybeId = null as Document['id'] | null;
+
+    const dispatch = createEventDispatcher();
+    let camStart = false as boolean;
+    let qrScanner = null as QrScanner | null;
+    let videoElement = null as HTMLVideoElement | null;
+    let uploadElement = null as HTMLInputElement | null;
+
+    async function startCamera() {
+        camStart = true;
+        await qrScanner?.start();
+    }
+
+    function stopCamera() {
+        camStart = false;
+        return qrScanner?.stop();
+    }
+
+    async function handleFileInput() {
```

Comment:
Why not just use `this` so that we don't have to `bind:this` anymore?
```suggestion
    async function handleFileInput(this: HTMLInputElement) {
```